AprilSylph/Palettes-for-Tumblr
# Add new color variables to built-in non-tumblr palettes

### Description


The new Tumblr design system has a new (huge) set of theme color variables, which Palettes for Tumblr doesn't currently override (see #191). Still not sure what we want to do about user-generated palettes re: these new colors, as it seems impractical to ask palette authors to manually pick all of them (particularly when, from the outside, we have no clue what any of them are for).

I assume the vast majority of extension users are using it for the three built-in non-tumblr palettes, though, and for those we only have to do this once. This thus just becomes a simple matter of manually picking 500 colors. Easy, no big deal. This PR is the result of doing this.

Just kidding. I used generative AI.

Just kidding again. I reverse-engineered a close approximation of the algorithm used to generate every new color key in the native true blue palette based on the original palette colors, then applied that to our three non-tumblr palettes (this involved using the https://github.com/color-js/color.js interpolation feature in srgb mode). Only the pink colors required some manual touch-ups, since for whatever reason the "pink" color in these palettes is grey; I don't actually know if this is a bug?

I don't have a demo UI, per se, but here's a screenshot of my dev UI with the real true blue, simulated true blue, 2016, 2013, and decision palette colors rendered:

<img height="300" src="https://github.com/user-attachments/assets/46756868-5a3d-4249-9a47-8d6d374437ff">

### Testing steps


Select one of the built-in non-tumblr palettes and confirm that elements such as the create button and notification badges are themed correctly. Confirm that elements are themed correctly on recently created pages that use the new theme color variables such as:

- https://www.tumblr.com/search/xkit/communities
- https://www.tumblr.com/communities/test-community-46174/settings
- https://www.tumblr.com/communities/communities-feedback/about

One can press shift-P repeatedly to cycle the selected Tumblr palette; in an ideal world this would have no visual effect on any of these pages when an updated extension palette is selected. This isn't actually the case, since some components (premium perks button, a selected community in the sidebar, the text below the new post bar, etc) are coded with DOM changes based on the currently selected palette, but there should be fewer of these changes with this PR.
